docs/cmake-presets.md

```diff
@@ -0,0 +1,296 @@
+# Configure and build with CMake Presets in VS Code
+
+CMake supports two files, `CMakePresets.json` and `CMakeUserPresets.json`, that allow users to specify common configure, build, and test options and share them with others.
+
+`CMakePresets.json` and `CMakeUserPresets.json` can be used to drive CMake in Visual Studio and Visual Studio Code, in a Continuous Integration (CI) pipeline, and from the command line. `CMakePresets.json` is intended to save project-wide builds, and `CMakeUserPresets.json` is intended for developers to save their own local builds.
```

Comment:
Let's add one sentence to the end of this paragraph: `CMakePresets.json` integration is available in cmake-tools version 1.7 and later.

Reply:
done